src/utils.ts

```diff
@@ -114,46 +114,42 @@ export class Utils {
       return nodes.sort((b, a) => (b.x + b.y * column)-(a.x + a.y * column));
   }
 
-  /**
-   * creates a style sheet with style id under given parent
-   * @param id will set the 'gs-style-id' attribute to that id
-   * @param parent to insert the stylesheet as first child,
-   * if none supplied it will be appended to the document head instead.
-   */
-  static createStylesheet(id: string, parent?: HTMLElement): CSSStyleSheet {
-    let style: HTMLStyleElement = document.createElement('style');
-    style.setAttribute('type', 'text/css');
-    style.setAttribute('gs-style-id', id);
-    // eslint-disable-next-line @typescript-eslint/no-explicit-any
-    if ((style as any).styleSheet) { // TODO: only CSSImportRule have that and different beast ??
-      // eslint-disable-next-line @typescript-eslint/no-explicit-any
-      (style as any).styleSheet.cssText = '';
-    } else {
-      style.appendChild(document.createTextNode('')); // WebKit hack
-    }
-    if (!parent) {
-      // default to head
-      parent = document.getElementsByTagName('head')[0];
-      parent.appendChild(style);
-    } else {
-      parent.insertBefore(style, parent.firstChild);
+  /** update CSS style on elements with given selector */
+  static updateStyleOnElements(selector: string, styles: {[props: string]: string |  string[] }): void {
+    const elements = Utils.getElements(selector);
+    for (const el  of elements) {
+      Utils.addElementStyle(el, styles)
     }
-    return style.sheet as CSSStyleSheet;
   }
 
-  /** removed the given stylesheet id */
-  static removeStylesheet(id: string): void {
-    let el = document.querySelector('STYLE[gs-style-id=' + id + ']');
-    if (el && el.parentNode) el.remove();
+  /** update CSS style on element */
+  static addElementStyle(el: HTMLElement, styles: {[props: string]: string |  string[] }): void {
+    if (styles instanceof Object) {
+      for (const s in styles) {
+        if (styles.hasOwnProperty(s)) {
+          if (Array.isArray(styles[s])) {
+            // support fallback value
+            (styles[s] as string[]).forEach(val => {
+              el.style[s] = val;
+            });
+          } else {
+            el.style[s] = styles[s];
+          }
+        }
+      }
+    }
   }
 
-  /** inserts a CSS rule */
-  static addCSSRule(sheet: CSSStyleSheet, selector: string, rules: string): void {
-    if (typeof sheet.addRule === 'function') {
-      sheet.addRule(selector, rules);
-    } else if (typeof sheet.insertRule === 'function') {
-      sheet.insertRule(`${selector}{${rules}}`);
-    }
+  /** update position style on element */
+  static updatePositionStyleOnWidget(el: HTMLElement, getHeight:(rows: number) => string): void {
+    const h = Utils.toNumber(el.getAttribute('gs-h'));
+    const minH= Utils.toNumber(el.getAttribute('gs-min-h'));
+    const maxH = Utils.toNumber(el.getAttribute('gs-max-h'));
+    const y = Utils.toNumber(el.getAttribute('gs-y'));
+    el.style.height = getHeight(h);

```

Comment:
I would pass GridStackPosition instead.
and you are setting the height(s) and top only - what about left and widths ? seem one sided...

Reply:
also I would not set properties (like min/max) that are not set